### PR TITLE
config, variable: add memory quota config for a query

### PR DESCRIPTION
**Pull request by zz-jason**
Thank you for working on TiDB! Please read TiDB's [CONTRIBUTING](https://github.com/pingcap/tidb/blob/master/CONTRIBUTING.md) document **BEFORE** filing this PR.

## What have you changed? (mandatory)

Before this PR, the default memory quota for a query is 32GB and can not be configured, this PR:
1. add a config entry named **mem-quota-query** to config the default memory quota for a query.
2. set the default memory quota for a query from 32GB to 5GB
3. we can still change the session variable **tidb_mem_quota_query** to control the memory quota for a query

## What are the type of the changes (mandatory)?

- Improvement (non-breaking change which is an improvement to an existing feature)

## How has this PR been tested (mandatory)?

- unit test
- explain test

## Does this PR affect documentation (docs/docs-cn) update? (optional)

Yes

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -245,6 +246,7 @@ var defaultConf = Config{
 	Lease:           "45s",
 	TokenLimit:      1000,
 	OOMAction:       "log",
```
[the remark this answers is not in the document]
Comment (zz-jason) [reply config/config.go]: we already have one: **tidb_mem_quota_query**

**Comment by lysu** [comment]
LGTM

**Comment by coocood** [comment]
@zz-jason 
If the user upgrades TiDB, then some existing query may fail.
Why not use the same default value 32GB?
If memory needs to be limited, they can set it lower.

**Comment by coocood** [comment]
@zz-jason 
The existing query may use more than 5GB memory.

**Comment by zz-jason** [comment]
@coocood Yes, but I think they will not fail if we don’t change the oom-action to “fail”. This config only affects the default value of the memory quota for a query, which is a session variable, not a global variable.

**Comment by coocood** [comment]
LGTM

**Pull request merged**